punchbowl/levelq/flow.py

```diff
@@ -1,40 +1,57 @@
+import os
+import multiprocessing
 from datetime import UTC, datetime
-from collections.abc import Callable
 
 import numpy as np
 from astropy.nddata import StdDevUncertainty
 from ndcube import NDCube
 from prefect import flow, get_run_logger
 
-from punchbowl.data import NormalizedMetadata, get_base_file_name
+from punchbowl.data import NormalizedMetadata, get_base_file_name, load_ndcube_from_fits
 from punchbowl.data.meta import set_spacecraft_location_to_earth
 from punchbowl.data.wcs import load_quickpunch_mosaic_wcs, load_quickpunch_nfi_wcs
 from punchbowl.level2.merge import merge_many_clear_task
 from punchbowl.level2.resample import reproject_many_flow
 from punchbowl.levelq.pca import pca_filter
-from punchbowl.util import average_datetime, load_image_task, output_image_task
+from punchbowl.util import DataLoader, average_datetime, load_image_task, output_image_task
 
 ORDER_QP = ["CR1", "CR2", "CR3", "CNN"]
 
 @flow(validate_parameters=False)
 def levelq_CNN_core_flow(data_list: list[str] | list[NDCube], #noqa: N802
                          output_filename: list[str] | None = None,
-                         files_to_fit: list[str | NDCube | Callable] | None = None) -> list[NDCube]:
+                         files_to_fit: list[str | NDCube | DataLoader] | None = None,
+                         outlier_limits: str | None = None,
+                         data_root: str | None = None) -> list[NDCube]:
```

Comment:
We really should have a more detailed doc string that describes these things. I know you inherited it like this, but could you fix it?